progenitor-impl/src/lib.rs

```diff
@@ -492,38 +501,58 @@ impl Generator {
         })
     }
 
+    fn get_dep_version(&self, name: &str) -> String {
+        match self.settings.dependency_versions.get(&name.to_string()) {
+            Some(o) => o.to_string(),
+            None => "*".to_string(),
+        }
+    }
+
     pub fn dependencies(&self) -> Vec<String> {
```

Comment:
what if we change this to just enumerate a list of dependencies without version numbers and then leave it to the consumer (i.e. progenitor stand-alone) to fill in the version numbers? Then we can do away with any knowledge of version numbers in progenitor-impl (or progenitor-macro) where it really isn't relevant.

Reply:
I don't want to feature creep this PR any more than necessary. Let's just let the features for reqwest etc be implicit and we can sort out this interaction later.

Reply:
That would be a regression of https://github.com/oxidecomputer/progenitor/pull/232 . I'd rather we sort it out this interaction now.

Reply:
I'm not sure how that follows, but, ok. You're proposing we replace `dependencies -> Vec` with `uses_X -> bool` for the various optional dependencies? Fine.